src/index.js

```diff
@@ -520,6 +520,56 @@ let Match = ({ path, children }) => (
   </BaseContext.Consumer>
 );
 
+////////////////////////////////////////////////////////////////////////////////
+// Hooks
+
+const useLocation = () => {
+  const context = useContext(LocationContext);
+
+  if (!context) {
+    throw new Error(
+      "useLocation hook was used but a LocationContext.Provider was not found in the parent tree. Make sure this is used in a component that is a child of Router"
+    );
+  }
+
+  return context.location;
+};
+
+const useNavigate = () => {
+  const context = useContext(LocationContext);
```

Comment:
Shouldn't all hooks have the `if (!context) {` guard?

Possibly by just having an internal `useLocationContext` and `useBaseContext` that are not exported to keep it a bit more DRY

Reply:
Yeah why not. the above hook was pseudo-copied from an above render prop component. It's unlikely useNavigate and other hooks will be used in similar issues, but it doesn't hurt to add helpful warnings when tools are misconfigured. Happy to take a PR to fix this!

Reply:
Never mind, I took care of it! Thanks for the call out